std-bits/table/src/main/java/org/enso/table/data/table/join/JoinResult.java

```diff
@@ -0,0 +1,7 @@
+package org.enso.table.data.table.join;
+
+import org.graalvm.collections.Pair;
+
+import java.util.List;
+
+public record JoinResult(List<Pair<Integer, Integer>> matchedRows) {}
```

Comment:
I wonder if we would be better with

```suggestion
public record JoinResultRow(int leftRow, int rightRow) {}

public record JoinResult(List<JoinResultRow> matchedRows) {}
```